mmdet/datasets/pipelines/transforms.py

```diff
@@ -1809,3 +1809,271 @@ def __repr__(self):
+                                               crop_w=self.size[1] - cut_x,
+                                               crop_h=cut_y,
+                                               mix_x=cut_x,
+                                               mix_y=0)
+
+            ignores, _ = self.filter_truth(results_i['gt_bboxes_ignore'],
+                                           None,
+                                           crop_w=self.size[1] - cut_x,
+                                           crop_h=cut_y,
+                                           mix_x=cut_x,
+                                           mix_y=0)
+            tmp_img[:cut_y, cut_x:] = results_i['img'].copy()
+        elif i == 2:
+            # Image 2: bottom left
+            results_i = self._crop_data(results_i, (self.size[0] - cut_y, cut_x), self.allow_negative_crop)
+            bboxes, labels = self.filter_truth(results_i['gt_bboxes'],
+                                               results_i['gt_labels'],
+                                               crop_w=cut_x,
+                                               crop_h=self.size[0] - cut_y,
+                                               mix_x=0,
+                                               mix_y=cut_y)
+            ignores, _ = self.filter_truth(results_i['gt_bboxes_ignore'],
+                                           None,
+                                           crop_w=cut_x,
+                                           crop_h=self.size[0] - cut_y,
+                                           mix_x=0,
+                                           mix_y=cut_y)
+            tmp_img[cut_y:, :cut_x] = results_i['img'].copy()
+        elif i == 3:
+            results_i = self._crop_data(results_i, (self.size[0] - cut_y, self.size[1] - cut_x),
+                                        self.allow_negative_crop)
+            # Image 3: bottom right
+            bboxes, labels = self.filter_truth(results_i['gt_bboxes'],
+                                               results_i['gt_labels'],
+                                               crop_w=self.size[1] - cut_x,
+                                               crop_h=self.size[0] - cut_y,
+                                               mix_x=cut_x,
+                                               mix_y=cut_y)
+            ignores, _ = self.filter_truth(results_i['gt_bboxes_ignore'],
+                                           None,
+                                           crop_w=self.size[1] - cut_x,
+                                           crop_h=self.size[0] - cut_y,
+                                           mix_x=cut_x,
+                                           mix_y=cut_y)
+            tmp_img[cut_y:, cut_x:] = results_i['img'].copy()
+
+        return tmp_img, bboxes, labels, ignores
+
+    def _crop_data(self, results, crop_size, allow_negative_crop):
+        """Function to randomly crop images, bounding boxes, masks, semantic
```

Comment:
We may maintain a `RandomCrop` object and add a `set_crop_size` method in this class. Therefore, this `_crop_data` method is not necessary any more.

Reply:
Add support for segmentation.